compiler/rustc_lint/src/builtin.rs

```diff
@@ -2278,6 +2278,21 @@ impl EarlyLintPass for IncompleteInternalFeatures {
                     );
                 }
             });
+
+        for cnum in [LOCAL_CRATE].into_iter().chain(cx.tcx.crates(()).iter().copied()) {
+            let crate_features = cx.tcx.lib_features(cnum);
+            for &(name, span) in &features.declared_lib_features {
+                if let Some((is_unstable, _)) = crate_features.unstable.get(&name) {
+                    if *is_unstable {
+                        cx.emit_spanned_lint(
+                            INTERNAL_FEATURES,
+                            span,
+                            BuiltinInternalFeatures { name, note: None },
+                        );
+                    }
+                }
+            }
+        }
```

Comment:
Should all this method go in `rustc_passes::stability::check_unused_or_stable_features`?